src/Uno.UI/Microsoft/UI/Xaml/Controls/TeachingTip/TeachingTip.cs

```diff
@@ -0,0 +1,2565 @@
+			}
+		}
+
+		private void UpdateDynamicHeroContentPlacementToBottom()
+		{
+			if (HeroContentPlacement == TeachingTipHeroContentPlacementMode.Auto)
+			{
+				UpdateDynamicHeroContentPlacementToBottomImpl();
+			}
+		}
+
+		private void UpdateDynamicHeroContentPlacementToBottomImpl()
+		{
+			VisualStateManager.GoToState(this, "HeroContentBottom", false);
+			if (m_currentHeroContentEffectivePlacementMode != TeachingTipHeroContentPlacementMode.Bottom)
+			{
+				m_currentHeroContentEffectivePlacementMode = TeachingTipHeroContentPlacementMode.Bottom;
+				TeachingTipTestHooks.NotifyEffectiveHeroContentPlacementChanged(this);
+			}
+		}
+
+		private void OnIsOpenChanged()
+		{
+			SharedHelpers.QueueCallbackForCompositionRendering(() =>
+			{
+				if (this.IsOpen)
+				{
+					this.IsOpenChangedToOpen();
+				}
+				else
+				{
+					this.IsOpenChangedToClose();
+				}
+				TeachingTipTestHooks.NotifyOpenedStatusChanged(this);
+			});
+		}
+
+		private void IsOpenChangedToOpen()
+		{
+			//Reset the close reason to the default value of programmatic.
+			m_lastCloseReason = TeachingTipCloseReason.Programmatic;
+
+			m_currentBoundsInCoreWindowSpace = this.TransformToVisual(null).TransformBounds(new Rect(
+				0.0,
+				0.0,
+				(float)(this.ActualWidth),
+				(float)(this.ActualHeight)));
+
+			Rect GetCurrentTargetBoundsInCoreWindowSpace()
+			{
+				var target = m_target;
+				if (target != null)
+				{
+					SetViewportChangedEvent(target);
+					return target.TransformToVisual(null).TransformBounds(new Rect(
+						0.0,
+						0.0,
+						(float)(target.ActualWidth),
+						(float)(target.ActualHeight)));
+				}
+				return Rect.Empty;
+			}
+
+			m_currentTargetBoundsInCoreWindowSpace = GetCurrentTargetBoundsInCoreWindowSpace();
+
+			if (m_lightDismissIndicatorPopup == null)
+			{
+				CreateLightDismissIndicatorPopup();
+			}
+			OnIsLightDismissEnabledChanged();
+
+			if (m_contractAnimation == null)
+			{
+				CreateContractAnimation();
+			}
+			if (m_expandAnimation == null)
+			{
+				CreateExpandAnimation();
+			}
+
+			// We are about to begin the process of trying to open the teaching tip, so notify that we are no longer idle.
+			SetIsIdle(false);
+
+			//If the developer defines their TeachingTip in a resource dictionary it is possible that it's template will have never been applied
+			if (!m_isTemplateApplied)
+			{
+				this.ApplyTemplate();
+			}
+
+			if (m_popup == null || m_createNewPopupOnOpen)
+			{
+				CreateNewPopup();
+			}
+
+			// If the tip is not going to open because it does not fit we need to make sure that
+			// the open, closing, closed life cycle still fires so that we don't cause apps to leak
+			// that depend on this sequence.
+			var (ignored, tipDoesNotFit) = DetermineEffectivePlacement();
+			if (tipDoesNotFit)
+			{
+				//__RP_Marker_ClassMemberById(RuntimeProfiler.ProfId_TeachingTip, RuntimeProfiler.ProfMemberId_TeachingTip_TipDidNotOpenDueToSize);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6388179)